bool clientConnection
bool thread
bool bufferReceived
